tests/unit/s2n_early_data_io_test.c

```diff
@@ -149,8 +149,14 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(server_conn, "test_all_tls12"));
             EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config_with_cert));
 
-            EXPECT_FAILURE_WITH_ERRNO(s2n_negotiate_test_server_and_client(server_conn, client_conn),
-                    S2N_ERR_PROTOCOL_VERSION_UNSUPPORTED);
+            /* Stop the TLS1.2 server before the CLIENT_KEY message. At that point, it will attempt to read
```

Comment:
Why does it receive the CCS message? I'm just wondering if this is exposing another error, or maybe the client here isn't set to tls1.2 or something?

Reply:
It receives the CCS message because 516a99e moved the Client CCS message from the second flight to the first.

The client here ISN'T set to TLS1.2; that's what we're testing. The test is to verify that when a TLS1.3 client sending early data receives a TLS1.2 ServerHello, it aborts the connection. It does that because a TLS1.2 server won't be able to handle the reordering of messages used to request early data, including the sending the client CCS early.

See https://tools.ietf.org/rfc/rfc8446#appendix-D.3

Reply:
Previously:
Client write ClientHello, Server read ClientHello, Server write \<lots of server messages\>, Server block, Client read ServerHello, Client fail with expected error relevant to test

After merge conflict:
Client write ClientHello, **Client write CCS,** Server read ClientHello, Server write \<lots of server messages\>, **Server read CCS, Server fail with expected error irrelevant to test. We never get to:** Client read ServerHello, Client fail with expected error